webhook/main.go

```diff
@@ -28,23 +33,59 @@ func main() {
 		redisPassword = "" // Default password (empty in this case)
 	}
 
-	client := redis.NewClient(&redis.Options{
-		Addr:     redisAddress,
-		Password: redisPassword,
-		DB:       0, // Default DB
-	})
+	client, err := createRedisClient()
+	if err != nil {
+		log.Fatalf("Failed to create Redis client: %v", err)
+	}
+	log.Println(redisPassword, redisAddress)
+
 	// Create a channel to act as the queue
 	webhookQueue := make(chan redisClient.WebhookPayload, 100) // Buffer size 100
 
 	go queue.ProcessWebhooks(ctx, webhookQueue)
 
 	// Subscribe to the "transactions" channel
-	err := redisClient.Subscribe(ctx, client, webhookQueue)
+	err = redisClient.Subscribe(ctx, client, webhookQueue)
 
 	if err != nil {
 		logging.WebhookLogger(logging.ErrorType, fmt.Errorf("error initializing connection: %s", err))
+		log.Fatalf("error initializing connection: %v", err)
+		return
 	}
 
 	select {}
+}
+
+func createRedisClient() (*redis.Client, error) {
+	redisAddress := os.Getenv("REDIS_ADDRESS")
+	if redisAddress == "" {
+		redisAddress = "localhost:6379" // Default address
+	}
+
+	redisPassword := os.Getenv("REDIS_PASSWORD")
+
+	// SSL/TLS configuration
+	useSSL := strings.ToLower(os.Getenv("REDIS_SSL")) == "true"
+	var tlsConfig *tls.Config
+
+	if useSSL {
+		caCertPath := os.Getenv("REDIS_CA_CERT")
+		clientCertPath := os.Getenv("REDIS_CLIENT_CERT")
+		clientKeyPath := os.Getenv("REDIS_CLIENT_KEY")
+
+		var err error
+		tlsConfig, err = redis_tls_config.CreateTLSConfig(caCertPath, clientCertPath, clientKeyPath)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	client := redis.NewClient(&redis.Options{
+		Addr:      redisAddress,
+		Password:  redisPassword,
+		DB:        0,
```

Comment:
Not always, add this one to func Params with 0 as a default value

Reply:
Alright